crates/project/src/lsp_store/semantic_tokens.rs

```diff
@@ -0,0 +1,104 @@
+use std::slice::ChunksExact;
+
+use crate::lsp_command::SemanticTokensEdit;
+
+/// All the semantic tokens for a buffer, in the LSP format.
+#[derive(Default, Debug, Clone)]
+pub struct SemanticTokens {
+    /// Each value is:
+    /// data[5*i] - deltaLine: token line number, relative to the start of the previous token
```

Comment:
Feels worth a few unit test around this logic (maybe, on a level above, but somewhere, we do have to test delta and other cases, e.g. the one described in a [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens)?)

Reply:
Thank you for adding tests both on the LSP part the "unit" tests on logic from this module.

Yet, I have failed to find any "unit" tests on applying delta requests on top of the partial one, and the spec seem to describe some example?

That looked somewhat complex to me, and if I'm not confusing things, would be great to add another test.